scrapy/utils/misc.py

```diff
@@ -226,7 +226,14 @@ def returns_none(return_node):
         return value is None or isinstance(value, ast.NameConstant) and value.value is None
 
     if inspect.isgeneratorfunction(callable):
-        code = re.sub(r"^[\t ]+", "", inspect.getsource(callable))
+        pattern = r"(^[\t ]+)"
```

Comment:
Can this pattern be compiled? It wasn't in the original version, but I guess it could mean a slight performance improvement.